### PR TITLE
Allow using ChunkedForest in SharedTree

### PR DESCRIPTION
**Pull request by daesunp**
## Description

This PR adds the option of using `ChunkedForest` in `SharedTree`

### CODE REVIEW

File: experimental/dds/tree2/src/shared-tree/sharedTree.ts [hunk 1]
```diff
@@ -238,7 +243,27 @@ export class SharedTree
 /**
  * @alpha
  */
-export interface SharedTreeOptions extends Partial<ICodecOptions> {}
+export interface SharedTreeOptions extends Partial<ICodecOptions> {
+	/**
+	 * The {@link ForestType} indicating which forest type should be created for the SharedTree.
+	 */
+	forest?: ForestType;
```
Comment (CraigMacomber) [inline experimental/dds/tree2/src/shared-tree/sharedTree.ts:250]: The default is not stated here, nor is it explicit in the implementation where this is used.

I good pattern I like for this kind of things is to export const that has the defaults (ex: const defaultOptions: Required<SharedTreeOptions> = {......};`

Then when used in Shared tree you can do: `const fullOptions = {...defaultOptions, ...options};

I think that approach scales better to more options, and makes it easy to change what the defaults are in a single place.

File: experimental/dds/tree2/src/test/shared-tree/sharedTree.spec.ts [hunk 2]
```diff
@@ -1919,6 +1920,41 @@ describe("SharedTree", () => {
...
+	describe("Creates a SharedTree using specific ForestType", () => {
+		it("unspecified ForestType uses ObjectForest", () => {
+			const { trees } = new TestTreeProviderLite(
+				1,
+				new SharedTreeFactory({
+					jsonValidator: typeboxValidator,
+					forest: ForestType.Reference,
+				}),
+			);
+			assert.equal(trees[0].forest.computationName, "object-forest.ObjectForest");
```
Comment (CraigMacomber) [inline experimental/dds/tree2/src/test/shared-tree/sharedTree.spec.ts:1933]: I didn't realize that would be a way to do this test. Neat trick, and keeps the test simple.

**Review by CraigMacomber — APPROVED** [review]

File: experimental/dds/tree2/src/shared-tree/sharedTree.ts (hunk 1 above)
Comment (daesunp) [reply experimental/dds/tree2/src/shared-tree/sharedTree.ts:250]: Added and exported `defaultSharedTreeOptions` to be used in sharedTree

File: experimental/dds/tree2/src/test/shared-tree/sharedTree.spec.ts (hunk 2 above)
Comment (daesunp) [reply experimental/dds/tree2/src/test/shared-tree/sharedTree.spec.ts:1932]: :)

**Pull request merged**